### PR TITLE
Added StringComparison to StringAssert Contains(), EndsWith() & StartsWith()

### LINKED ISSUE #154: StringAssert APIs do not have ignorecase overloads
## Description
Like Assert.AreEqual APIs StringAssert does not have APIs that Ignore case in the default set. Although it is possible to add an extension through the Assertion extensibility [here](https://github.com/Microsoft/testfx-docs/blob/master/RFCs/002-Framework-Extensibility-Custom-Assertions.md), this is more of an ask to include this in the default set of APIs.

### PR DESCRIPTION
**Pull request by HeroMaxPower**
Description:

Added StringComparison to StringAssert Contains, EndsWith & StartsWith
Added tests to test functionality
Added API comment on StringComparison Enum
Updated APIs to default to StringComparison.Ordinal 

Probably needs API documentation updates if this gets merged.

Resolves #154

### CODE REVIEW

File: src/TestFramework/MSTest.Core/Assertions/StringAssert.cs [hunk 1]
```diff
@@ -107,18 +158,21 @@ public static void Contains(string value, string substring, string message)
...
+        /// The comparison method to compare strings <paramref name="comparisonType"/>.
+        /// </param>
         /// <param name="parameters">
         /// An array of parameters to use when formatting <paramref name="message"/>.
         /// </param>
         /// <exception cref="AssertFailedException">
         /// Thrown if <paramref name="substring"/> is not found in
         /// <paramref name="value"/>.
         /// </exception>
-        public static void Contains(string value, string substring, string message, params object[] parameters)
```
Comment (304NotModified) [inline src/TestFramework/MSTest.Core/Assertions/StringAssert.cs:117]: please note, `Contains(string value, string substring, string message, params object[] parameters)` is now gone, so this is breaking change. 

Please also add  `Contains(string value, string substring, string message, params object[] parameters)` back

File: src/TestFramework/MSTest.Core/Assertions/StringAssert.cs [hunk 2]
```diff
@@ -88,7 +111,35 @@ public static void Contains(string value, string substring)
...
+        /// <param name="comparisonType">
+        /// The comparison method to compare strings <paramref name="comparisonType"/>.
+        /// </param>
+        /// <exception cref="AssertFailedException">
+        /// Thrown if <paramref name="substring"/> is not found in
+        /// <paramref name="value"/>.
+        /// </exception>
+        public static void Contains(string value, string substring, string message, StringComparison comparisonType)
+        {
+            Contains(value, substring, message, comparisonType, null);
```
Comment (304NotModified) [inline src/TestFramework/MSTest.Core/Assertions/StringAssert.cs]: are we sending `null` to ` params object[] parameters`? That isn't needed nor recommend? 

(same for StartsWith and EndsWith)

**Comment by nohwnd** [comment]
@HeroMaxPower are you going to address the PR comments or should I?

**Comment by HeroMaxPower** [comment]
Yes please, i haven't had time in my personal life to look at these. 👍

File: src/TestFramework/MSTest.Core/Assertions/StringAssert.cs (hunk 1 above)
Comment (nohwnd) [reply src/TestFramework/MSTest.Core/Assertions/StringAssert.cs:117]: Added

File: src/TestFramework/MSTest.Core/Assertions/StringAssert.cs (hunk 2 above)
Comment (nohwnd) [reply src/TestFramework/MSTest.Core/Assertions/StringAssert.cs]: added

**Comment by nohwnd** [comment]
@304NotModified I had to pass empty array to disambiguate the calls (before we passed null). The compiler will pass empty array for us, so we should not be doing anything wrong. Unless the user passed StringComparison as the actual first param, then they now would get different overload than before. But that is unlikely scenario.

**Comment by HeroMaxPower** [comment]
@304NotModified any updates?

**Review by 304NotModified — APPROVED** [review]
Looks good to me! Thanks for the reminder!

**Comment by HeroMaxPower** [comment]
@nohwnd i think this is ready to be merged 👍

**Comment by nohwnd** [comment]
@Haplois could you review and merge please.

**Review by Haplois — APPROVED** [review]

**Pull request merged**